foyer-common/src/queue.rs

```diff
@@ -16,18 +16,16 @@ use std::{
     fmt::Debug,
     sync::atomic::{AtomicUsize, Ordering},
 };
+use std::collections::VecDeque;
+use parking_lot::Mutex;
+use tokio::sync::Notify;
 
-use tokio::sync::{
-    mpsc::{unbounded_channel, UnboundedReceiver, UnboundedSender},
-    Mutex,
-};
 
 #[derive(Debug)]
-pub struct AsyncQueue<T: Debug> {
-    tx: UnboundedSender<T>,
-    rx: Mutex<UnboundedReceiver<T>>,
-
+pub struct AsyncQueue<T> {
+    queue: Mutex<VecDeque<T>>,
     size: AtomicUsize,
```

Comment:
If `Mutex` is used on `VecQueue` for both `acqiure` and `release` method, `size: AtomicUsize` can be removed and use `RwLock` on `VecQueue` then use shared lock for `len`

Reply:
fixed